examples/rate_smoothing.rs

```diff
@@ -0,0 +1,218 @@
+#![warn(clippy::pedantic)]
```

Comment:
I'd prefer not to add an example for this right now.

Reply:
this will be removed last, let's just make sure everything else is good first

Reply:
Let's remove this, you can just keep it in your local working dir for testing.

Reply:
I'm going to keep this until everything else looks good. The reason why is because I want to continually evaluate whether the syntax I'm able to use looks correct and idiomatic. It acts as a sanity check, and when I refactor anything crate side, it auto-refactors on my side as well

Reply:
Yes: which is why I suggested you keep this in your local clone without committing it into your Git commits. This should support your refactoring work without me having to look at the changes, and, since I am close to merging this, have to do an extra round trip to ask you to remove it.